test/deploy.py

```diff
@@ -0,0 +1,569 @@
+                        print(
+                            "Deleting Volume {} from Pool {}"
+                            .format(volume, pool_name)
+                        )
+                        pool.volumes.get("custom", volume).delete()
+
+                # Delete Storage pool
+                print("Deleting Pool {}".format(pool_name))
+                pool.delete()
+
+
+class DeployRunner:
+    # LXD Vars
+    deploy_tag = "oci-testing-" + _get_random_string(4)
+    complete_repo_path = "/home/ubuntu"
+    cwd = os.getcwd()
+    usr = pwd.getpwuid(os.getuid())[0]
+    modelFilePath = ""
+    model = dict()
+
+    def __init__(self) -> None:
+        # Check LXD installed on host.
+        self.check_snaps_installed()
+        # Check if current user is part of the lxd user group.
+        self.check_user_in_group()
+        # init client
+        self.client = pylxd.Client()
+        # Check if LXD is initialised.
+        self.check_lxd_initialised()
+        # File to store LXD virtual resource references.
+        self.modelFilePath = "{}/model-{}.json".format(
+            self.cwd, _get_random_string(4)
+        )
+
+    def save_model_json(self):
+        """Save vm resource dictionary to json file."""
+        with open(self.modelFilePath, "w") as model_file:
+            model_file.write(json.dumps(self.model, indent=4))
+            print("Model information exported to {}"
+                  .format(self.modelFilePath))
+
+    def check_snaps_installed(self, required_snaps: list = []):
+        """Check if snap dependencies are met."""
+        if not required_snaps:
+            required_snaps = ["lxd"]
+
+        cmd = ["snap", "list"]
+        output = subprocess.check_output(cmd).decode()
+        snaps = list(map(
+                lambda snap_entry: snap_entry.split(" ")[0],
+                output.splitlines()
+            ))
+
+        if not all(snap in snaps for snap in required_snaps):
+            raise AssertionError("Required snaps not installed: {}"
+                                 .format(snaps))
+
+    def check_user_in_group(self, group_name="lxd"):
+        """Check if current user belongs to provided user group"""
+        cmd = [
+            "getent",
+            "group",
+            group_name,
+        ]
+        output = subprocess.check_output(cmd).decode()
+
+        if not (self.usr in output and group_name in output):
+            raise AssertionError(
+                "User {} is not in group {}."
+                "\nnewgrp {}"
+                "\nsudo usermod -aG {} {}".format(
+                    self.usr, group_name, group_name, self.usr, group_name
+                )
+            )
+
+    def check_lxd_initialised(self) -> None:
+        """Check if LXD default profile contains network and root device."""
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        default_devices = ["eth0", "root"]
```

Comment:
Does it make sense to check for `eth0` ? I think having the network device doesn't really say much about _connectivity_, does it ?

Reply:
eth0 and root are the 2 default devices created by the lxd init process. Checking for these 2 (referred https://discuss.linuxcontainers.org/t/how-to-check-if-lxd-init-auto-already-ran/1635/) we can confirm if lxd init has already been done.